src/utils/rfqm_blockchain_utils.ts

```diff
@@ -0,0 +1,61 @@
+import { IZeroExContract } from '@0x/contracts-zero-ex';
+import { MetaTransaction, RfqOrder, Signature } from '@0x/protocol-utils';
+import { BigNumber } from '@0x/utils';
+
+import { NULL_ADDRESS } from '../constants';
+
+// allow a wide range for gas price for flexibility
+const MIN_GAS_PRICE = new BigNumber(0);
+// 10K Gwei
+const MAX_GAS_PRICE = new BigNumber(1e13);
+
+export class RfqBlockchainUtils {
+    constructor(private readonly _exchangeProxy: IZeroExContract) {}
+
+    // for use when 0x API operator submits an order on-chain on behalf of taker
+    public generateMetaTransaction(
+        rfqOrder: RfqOrder,
+        signature: Signature,
+        taker: string,
+        takerAmount: BigNumber,
+        chainId: number,
+    ): MetaTransaction {
+        // generate call data for fillRfqOrder
+        const callData = this._exchangeProxy
+            .fillRfqOrder(rfqOrder, signature, takerAmount)
+            .getABIEncodedTransactionData();
+
+        return new MetaTransaction({
+            signer: taker,
+            sender: NULL_ADDRESS,
+            minGasPrice: MIN_GAS_PRICE,
+            maxGasPrice: MAX_GAS_PRICE,
+            expirationTimeSeconds: rfqOrder.expiry,
+            salt: new BigNumber(Date.now()),
+            callData,
+            // may need this to be variable to handle ETH -> ERC20
+            value: new BigNumber(0),
+            feeToken: rfqOrder.takerToken,
+            feeAmount: new BigNumber(0),
+            chainId,
+            verifyingContract: this._exchangeProxy.address,
+        });
+    }
+
+    public async validateMetaTransactionOrThrowAsync(
+        metaTx: MetaTransaction,
+        metaTxSig: Signature,
+        sender: string,
+    ): Promise<boolean> {
+        try {
+            await this._exchangeProxy.executeMetaTransaction(metaTx, metaTxSig).callAsync({ from: sender });
```

Comment:
Could we parse the return value and ensure that the actual call data in the metatransaction was successful?

Reply:
Is it worth hard-coding some other values during the `eth_call` (such as max gas used, gas price)? This will call the trade to rightfully revert even if the metatransaction is valid but something is funky with the worker.

Reply:
(Example, worker is out of ETH, will cause trade to revert)

Reply:
added some more flexibility